CMakeLists.txt

```diff
@@ -336,6 +336,10 @@ IF(${Kokkos_ENABLE_Cuda})
       SET(KOKKOSKERNELS_ENABLE_TPL_CUSPARSE ${TPL_ENABLE_CUSPARSE})
     ENDIF()
   ENDIF()
+
+  IF (NOT DEFINED KOKKOSKERNELS_ENABLE_TPL_MAGMA)  
```

Comment:
What is it you're trying to do here?  Are you trying to add MAGMA as an optional TPL dependency of kokkos-kernels?  What you're doing here is trying to treat MAGMA the same way as cuBLAS, but you can't do that -- CUDA automatically brings in cuBLAS, but it doesn't (at least I don't think it does) automatically bring in MAGMA.  MAGMA is a separate TPL, and it should always be optional.

Reply:
Or perhaps things have changed, and MAGMA now comes with CUDA by default?

Reply:
@mhoemmen As far as I know, MAGMA always requires CUDA. So, I am trying to treat MAGMA the same way as cuBLAS, as a BLAS extension with dense system solvers.

Reply:
@vqd8a I think what you have is other way ..

Reply:
@srajama1 I am not quite clear what you mean?

Reply:
The issue is the other way around.  MAGMA requires CUDA, yes, but CUDA does not require MAGMA.  CUDA does not come with MAGMA built in.  Thus, it needs to be possible to build with CUDA enabled, but MAGMA disabled.  In other words, MAGMA must be an optional library TPL dependency of kokkos-kernels.

Reply:
@mhoemmen Thanks for explaining. Now that I was reading the CMakeLists.txt more carefully, I realized I was not paying attention to this (just doing pattern matching). I will fix it.

Reply:
CMakeLists.txt has been updated with MAGMA logic.